monai/transforms/utility/array.py

```diff
@@ -624,11 +626,23 @@ def __call__(self, img: NdarrayOrTensor) -> NdarrayOrTensor:
         """
         img = convert_to_tensor(img, track_meta=get_track_meta())
         if self.dim is None:
+            if self.update_meta:
+                warnings.warn("update_meta=True is ignored when dim=None.")
             return img.squeeze()
+        dim = (self.dim + len(img.shape)) if self.dim < 0 else self.dim
         # for pytorch/numpy unification
-        if img.shape[self.dim] != 1:
-            raise ValueError(f"Can only squeeze singleton dimension, got shape {img.shape}.")
-        return img.squeeze(self.dim)
+        if img.shape[dim] != 1:
+            raise ValueError(f"Can only squeeze singleton dimension, got shape {img.shape[dim]} of {img.shape}.")
+        img = img.squeeze(dim)
+        if self.update_meta and isinstance(img, MetaTensor) and dim > 0 and len(img.affine.shape) == 2:
+            h, w = img.affine.shape
+            affine, device = img.affine, img.affine.device if isinstance(img.affine, torch.Tensor) else None
```

Comment:
Do we need some warning if the input image is not in axial orientation？

Reply:
No, this is about squeezing any one of the spatial dimensions, I'll make an example in the discussion.

Reply:
Hi @wyli, 
Maybe I didn't express myself very clearly, I know that `SqueezeDim` can squeeze in any one of the spatial dimensions, but if the user doesn't turn the image into the axial orientation, then the updated affine will also be incorrect. I'm not sure if the warning needs to be in this function, or maybe not.
```
original affine
tensor([[ -0.7422,   0.0000,   0.0000, 186.3210],
        [  0.0000,   0.0000,  -3.0000,  70.6580],
        [  0.0000,  -0.7422,   0.0000, 189.4130],
        [  0.0000,   0.0000,   0.0000,   1.0000]], dtype=torch.float64)
with orientation
tensor([[   0.7422,    0.0000, -192.9432],
        [   0.0000,    0.7422, -189.8512],
        [   0.0000,    0.0000,    1.0000]])
without orientation
tensor([[ -0.7422,   0.0000, 186.3210],
        [  0.0000,   0.0000, 189.4130],
        [  0.0000,   0.0000,   1.0000]], dtype=torch.float64)
```

Reply:
I'm not sure how to warn properly within this transform, we couldn't assume orientation here. But I'll add a determinant value check, if it's zero it's probably problematic.